src/main/java/com/aws/greengrass/detector/uploader/ConnectivityUpdater.java

```diff
@@ -94,16 +97,26 @@ boolean hasPortChanged(int port) {
         return this.defaultPort != port;
     }
 
-    private UpdateConnectivityInfoResponse updateConnectivityInfo(List<ConnectivityInfo> connectivityInfoItems) {
+    private UpdateConnectivityInfoResponse updateConnectivityInfo(List<ConnectivityInfo> connectivityInfoItems)
+            throws DeviceConfigurationException {
         if (connectivityInfoItems == null || connectivityInfoItems.isEmpty()) {
             return null;
         }
 
-        UpdateConnectivityInfoRequest updateConnectivityInfoRequest =
-                UpdateConnectivityInfoRequest.builder().thingName(Coerce.toString(deviceConfiguration.getThingName()))
-                        .connectivityInfo(connectivityInfoItems).build();
+        try (GreengrassV2DataClient client = clientFactory.getGreengrassV2DataClient()) {
```

Comment:
`fetchGreengrassV2DataClient`?

Reply:
fetchGreengrassV2DataClient isn't available for Nucleus <2.5.6, so can't use it here since this needs to be compatible with older Nucleus versions. We have to use the original getGreengrassV2DataClient and handle the fact that it may return null.